Java/LFU.java

```diff
@@ -0,0 +1,144 @@
+package lfu;
+import java.util.LinkedList;
+import java.util.ListIterator;
+import java.util.HashMap;
+
+/**
+ * @author salman
+ * @param <T>
+ */
+
+public class LFU<T> {
+
+	private int MAX_SIZE = 2;
+
+	private LinkedList< FrequencyList<T> > lfuList;
+	private HashMap<T, Pair<Integer, ListIterator<T> > > nodeFrequencyMap;
```

Comment:
If I were you, I would never use ListIterator directory.

I would probably make it Iterable<T>